haystack/components/websearch/searchapi.py

```diff
@@ -64,17 +79,25 @@ def to_dict(self) -> Dict[str, Any]:
     @classmethod
     def from_dict(cls, data: Dict[str, Any]) -> "SearchApiWebSearch":
         """
-        Deserialize this component from a dictionary.
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+                The deserialized component.
         """
         deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
         return default_from_dict(cls, data)
 
-    @component.output_types(documents=List[Document], links=List[str])
-    def run(self, query: str):
+    @component.output_types(documents=List[Document], links=Union[List[Document], List[str]])
+    def run(self, query: str) -> Dict[str, Union[List[Document], List[str]]]:
         """
-        Search the SearchApi API for the given query and return the results as a list of Documents and a list of links.
+        Uses [SearchApi](https://www.searchapi.io/) to search the web.
 
-        :param query: Query string.
+        :param query: Search query.
+        :returns: The documents with the content of the search results and the links to the search results.
+        :raises TimeoutError: If the request to the SearchApi API times out.
+        :raises SearchApiError: If an error occurs while querying the SearchApi API.
```

Comment:
We should use the standard we decided on as specified in #7074.

Reply:
Same thing for `SerperDevWebSearch.run()`.

Reply:
where am I violating it? 🙈

Reply:
I would think it's better if we stick to this style for returns:

```
        :returns: A dictionary with the following keys:
            - `documents`: Created Documents
```

The rest is good though.

Reply:
aaah, okay 🙌🏻